### ISSUE TITLE
[BUG] optimizely.isReady() always returns false in version 3.2.3

### ISSUE THREAD

**Issue by mlazari** [optimizely/react-sdk#300]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SDK Version

3.2.3

### Current Behavior

If I use a OptimizelyProvider that on first render has user undefined, then the `isReady()` method of the optimizely client will always return false, even if the user is set later.
It seems to be related to this change that added verification for isUserReady to isReady(): https://github.com/optimizely/react-sdk/commit/62a12ff9bd1180335135e77124d27b6f0966ca1d#diff-25d66d74617fe2e23d7946bd6e3ba95640ab1b9bc8947445d604fc271c7c1f12R447

`isUserReady` seems to be changed only once here - https://github.com/optimizely/react-sdk/blob/3.2.3/src/client.ts#L252 - after `userReadyPromise` is resolved here when `setUser` is called first time (e.g. from OptimizelyProvider) - https://github.com/optimizely/react-sdk/blob/3.2.3/src/client.ts#L410. If `setUser` was called with the default user (with id = null), then the promise gets resolved with `{ success: false }`, thus `isUserReady` will remain false and never change again.

### Expected Behavior

`isReady()` should return true at some point when the optimizely client is ready.

### Steps To Reproduce

1. Clone https://github.com/mlazari/OptimizelyRepro and make sure you are in the main branch
2. Copy the `.env.development` file to a new `.env.local` file in the root of the project and set your optimizely key in it
3. Run `npm install`
4. Run `npx expo prebuild --clean`
5. Run the app with `npx expo run:ios` (or `npx expo run:android`)
6. Tap on the "Re-check" button, see the "Optimizely Ready" field remains false. Try again by tapping on "Set user info", then again on "Re-check", see it still says "false"

https://github.com/user-attachments/assets/7f90556e-03eb-4d5e-8294-edc331dd7aa2

7. Close the app and metro bundler
8. Switch to branch 3.2.2 (which has `@optimizely/react-sdk` downgraded to 3.2.2)
9. Run `npm install`
10. Run `npx expo prebuild --clean`
11. Run the app with `npx expo run:ios` (or `npx expo run:android`)
12. Tap on the "Re-check" button, see the "Optimizely Ready" field changes to true

https://github.com/user-attachments/assets/ed5939b3-2a56-40e7-bf02-10af50c1cbb9

### Link

https://github.com/mlazari/OptimizelyRepro/blob/main/app/index.tsx

### Severity

Blocking development

### Workaround/Solution

Downgrading to 3.2.2 for now.

**Comment by junaed-optimizely** [comment]
Hey @mlazari , thanks for very detailed report. 

I gave it a look. `{success: false}`  happens  when ODP is explicitly disabled and user is set to default user. 

Expected behavior is - "User promise should not be resolved unless a valid user is set, when ODP is disabled". 

This is an edge case that has not been considered in the SDK for a while. v3.2.3 just exposed it.

I have created an internal ticket FSSDK-10980 to track this. Will get back to it shortly! 

Thanks again.

**Comment by mlazari** [comment]
@junaed-optimizely Thank you! I was also thinking it's related to ODP being disabled, but from my debugging it was not reaching the `this.odpExplicitlyOff` verification at https://github.com/optimizely/react-sdk/blob/3.2.3/src/client.ts#L377C9-L377C30 because it was returning earlier at https://github.com/optimizely/react-sdk/blob/3.2.3/src/client.ts#L374 because userContext was null. But I guess it was null because of ODP being disabled?

**Comment by fmacherey** [comment]
I have a similar issue which leaded me to this issue. I am using nextjs with the react sdk.

My client becomes ready and everything is fine. Then i change the page (no hard window reload, just nextjs linking), the client is ready, but then becomes unready again.

The logging just says, there was a timeout creating the optimizely instance, but no concrete reason why. 

As a fix i downgraded to 2.9.2 which is with the ODP stuff, and that seems to work now.

Leaving this note here, if anyone else stumbles over this.

**Comment by junaed-optimizely** [comment]
A new release [v3.2.4](https://github.com/optimizely/react-sdk/releases/tag/3.2.4) is out with the fix!

**Comment by junaed-optimizely** [comment]
@fmacherey could you please try with the latest release? 

 If the issue still persists, please open a new issue with the framework, lib versions and implementation details.
